app.rb

```diff
@@ -0,0 +1,167 @@
+    loop do
+      menu
+      option = gets.chomp
+      break if option == '7'
+
+      get_num option
+    end
+    puts 'Thank you for using our Library!'
+  end
+
+  def menu
+    puts
+    puts 'Please choose an option by entering a number'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def get_num(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'Please enter a number between 1 and 7'
+    end
+  end
+
+  def list_all_books
+    puts 'No books are created yet! Please use option 5 to create a books.' if @books.empty?
+
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 0.75
+  end
+
+  def list_all_people
+    puts 'No people are created yet! Please use option 4 to create a person.' if @people.empty?
+    @people.map { |person| puts "Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    sleep 0.75
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or teacher (2) [Input respective number]: '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Please input 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+
+    student = Student.new(@classroom, age, name, parent_permission)
+    @people << student
+
+    puts 'Student successfully created!'
+    sleep 0.75
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(specialization, name, age)
+    @people << teacher
+
+    puts 'Teacher successfully created!'
+    sleep 0.75
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+
+    puts 'Book successfully created!'
+    sleep 0.75
+  end
+end
+
+def create_rental
+  puts 'Select a book from the following list by number'
+  @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+  book_id = gets.chomp.to_i
+
+  puts 'Select a person from the following list by number (not id)'
+  @people.each_with_index do |person, index|
+    puts "#{index}) [#{person.classroom}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
```

Comment:
- [ ] When I try to list the teacher object for creating a rental, I get an error message about an undefined classroom method for a teach. Kindly fix that. I think what you meant here is `person.class`, not `person.classroom`.

  ![lockless-teacher](https://user-images.githubusercontent.com/36542986/154949006-46cb5a0b-8691-4cdd-94cb-3d2228926db0.jpg)